File: app/src/main/java/home/self/beerviewer_mvvm/view/splash/SplashModule.java
package home.self.beerviewer_mvvm.view.splash;

import dagger.Module;
import dagger.Provides;
import home.self.beerviewer_mvvm.data.source.BeerRepository;
import home.self.beerviewer_mvvm.di.ActivityScope;
import home.self.beerviewer_mvvm.rx.schedulers.BaseSchedulerProvider;

/**
 * Created by Tak on 2018. 4. 19..
 */

@Module
public class SplashModule {

    @Provides
    @ActivityScope
    SplashViewModel provideSplashViewModel(BeerRepository beerRepository, BaseSchedulerProvider schedulerProvider) {
        return  new SplashViewModel(beerRepository, schedulerProvider);
    }
}

File: app-kotlin/src/main/java/home/self/beerviewer_mvvm/app_kotlin/view/splash/SplashModule.kt
package home.self.beerviewer_mvvm.app_kotlin.view.splash

import androidx.lifecycle.ViewModel
import com.mashup.dutchmarket.di.key.ViewModelKey
import dagger.Binds
import dagger.Module
import dagger.multibindings.IntoMap

/**
 * Created by Tak on 2018. 4. 19..
 */

@Module(includes = arrayOf(SplashModule.ProvideModule::class))
internal interface SplashModule {

    @Module
    class ProvideModule {

    }

    @Binds
    @IntoMap
    @ViewModelKey(SplashViewModel.ViewModel::class)
    fun bindSplashViewModel(splashViewModel: SplashViewModel.ViewModel): ViewModel

}
